component/src/main/java/org/wso2/extension/siddhi/store/ignite/ApacheIgniteStore.java

```diff
@@ -0,0 +1,1158 @@
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "distribute.joins",
+                        description = "Whether to use distributed joins for non collocated data or not. ",
+                        optional = true,
+                        defaultValue = "false",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "enforce.join.order",
+                        description = "Whether to enforce join order of tables in the query or not. If set to true" +
+                                " query optimizer will not reorder tables in join. ",
+                        optional = true,
+                        defaultValue = "false ",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "collocated",
+                        description = "Whether your data is co-located or not ",
+                        optional = true,
+                        defaultValue = "false",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "replicated.only",
+                        description = "Whether query contains only replicated tables or not ",
+                        optional = true,
+                        defaultValue = "false",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "auto.close.server.cursor",
+                        description = "Whether to close server-side cursor automatically when last piece of " +
+                                "result set is retrieved or not. ",
+                        optional = true,
+                        defaultValue = "false",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "socket.send.buffer",
+                        description = "Socket send buffer size.When set to 0, OS default will be used. ",
+                        optional = true, defaultValue = "0 ",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "socket.receive.buffer",
+                        description = "Socket receive buffer size.When set to 0, OS default will be used. ",
+                        optional = true,
+                        defaultValue = "0",
+                        type = {DataType.STRING}
+                ),
+
+                @Parameter(
+                        name = "backups",
+                        description = "Number of backup copies of data.It can take the value of any positive integer",
+                        optional = true,
+                        defaultValue = "0",
+                        type = {DataType.STRING}
```

Comment:
Please double check all the parameter type. You can use INT, BOOLEAN according to your requirements instead of using STRING.